### PR TITLE
add options params support for WS

### PR DESCRIPTION
**Pull request by timCF**
now ws can accept options like timeout etc

### CODE REVIEW

File: lib/socket/web.ex [hunk 1]
```diff
@@ -271,9 +271,7 @@ defmodule Socket.Web do
   Listens on the default port (80).
   """
   @spec listen :: { :ok, t } | { :error, error }
-  def listen do
-    listen([])
-  end
+  def listen, do: listen([])
```
Comment (meh) [inline lib/socket/web.ex]: Please avoid inline `do`s.

File: mix.exs [hunk 2]
```diff
@@ -4,7 +4,7 @@ defmodule Socket.Mixfile do
   def project do
     [ app: :socket,
       version: "0.3.0",
-      elixir: "~> 1.0.0-rc1",
+      #elixir: "~> 1.1.1",
```
Comment (meh) [inline mix.exs]: A commit with mix stuff snuck in, please remove it so I can do the version upgrade myself (including publishing).

File: lib/socket/web.ex [hunk 3]
```diff
@@ -661,7 +658,7 @@ defmodule Socket.Web do
                 opcode :: 4,
                 mask   :: 1,
                 length :: 7 >> } when known?(opcode) and control?(opcode) ->
-        on_success case opcode(opcode), do: (
+        on_success((case opcode(opcode), do: (
```
Comment (meh) [inline lib/socket/web.ex]: This became really ugly, do you think there's a way to avoid it?

**Comment by timCF** [comment]
done :)

**Comment by meh** [comment]
Merged, thanks! :panda_face:

**Pull request closed without merging**